src/Appwrite/Platform/Tasks/Backup.php

```diff
@@ -0,0 +1,252 @@
+
+use Exception;
+use Utopia\DSN\DSN;
+use Utopia\Platform\Action;
+use Utopia\App;
+use Utopia\CLI\Console;
+use Utopia\Pools\Group;
+use Utopia\Storage\Device;
+use Utopia\Storage\Device\DOSpaces;
+use Utopia\Storage\Device\Local;
+use Utopia\Validator\Text;
+
+class Backup extends Action
+{
+    protected ?DSN $dsn = null;
+    protected ?string $database = null;
+    public const BACKUPS = '/backups';
+    public const BACKUP_INTERVAL = 60 * 60 * 4; // 4 hours;
+    public const COMPRESS_ALGORITHM = 'lz4';
+    public const CNF = '/etc/my.cnf';
+    public const PROCESSORS = 4;
+
+    public static function getName(): string
+    {
+        return 'backup';
+    }
+
+    public function __construct()
+    {
+        $this->checkEnvVariables();
+
+        $this
+            ->desc('Backup a DB')
+            ->param('database', null, new Text(20), 'passed from command')
+            ->inject('pools')
+            ->callback(fn(string $database, Group $pools) => $this->action($database, $pools));
+    }
+
+    /**
+     * @throws Exception
+     */
+    public function action($database, Group $pools): void
+    {
+        $this->database = $database;
+        $this->dsn = self::getDsn($database);
+
+        if (!$this->dsn instanceof DSN) {
+            Console::error('No dsn match');
+            Console::exit();
+        }
+
+        $attempts = 0;
+        $max = 10;
+        $sleep = 5;
+
+        do {
+            try {
+                $attempts++;
+                $pools
+                    ->get('database_' . $database)
+                    ->pop()
+                    ->getResource()
+                ;
+
+                break; // leave the do-while if successful
+            } catch (Exception $e) {
+                Console::warning("Database not ready. Retrying connection ({$attempts})...");
+                if ($attempts >= $max) {
+                    throw new Exception('Failed to connect to database: ' . $e->getMessage());
+                }
+
+                sleep($sleep);
+            }
+        } while ($attempts < $max);
+
+        Console::loop(function () {
+            $this->start();
+        }, self::BACKUP_INTERVAL);
+    }
+
+    public function start(): void
+    {
+        $start = microtime(true);
+        $time = date('Y_m_d_H_i_s');
+
+        self::log('--- Backup Start --- ');
+        self::log('--- Creating backup ' . $time . '  --- ');
+
+        $filename = $time . '.tar.gz';
+        $local = new Local(self::BACKUPS . '/' . $this->database . '/full/' . $time);
+        $local->setTransferChunkSize(5  * 1024 * 1024); // > 5MB
+
+        $backups = $local->getRoot() . '/files';
+        $tarFile = $local->getPath($filename);
+
+        $this->fullBackup($backups);
+        $this->tar($backups, $tarFile);
+        $this->upload($tarFile, $local);
+        // todo: Do we want to delete the tar file? and remain with the folder?
+        // todo: Do we want to delete the backup.log?
```

Comment:
Yep let's delete them from the local machine as long as the upload is successful

Reply:
Done!